### DISCUSSION TITLE
Access Static Member of an Generic Type Parameter

### DISCUSSION THREAD

**Discussion by inlineHamed** [dotnet/csharplang#5112] [Language Ideas]
It maybe a duplicate, I searched for any history a while but seems to be hard in this case to find if any history exists so I decide to open a new item.

I was generalizing some part of my code that reach a limitation that seems simple at first but it seems that there is no clean way for that (I found a similar need here where no straight way answer is gave: https://stackoverflow.com/questions/19843/generics-in-c-sharp-accessing-the-static-members-of-t)

Here's a sample code:

```
abstract class Base {
   public abstract static string X {get;}
}
class A : Base {
   public static string X => "X in A";
}

class GenericClass<T> where T: Base {
   void SomeMethod(){
      Console.Out.WriteLine(T.X);
   }
}
```

I think it would be nice if C# can handle this case.

**Comment by FaustVX** [comment]
- Static members can't be `abstract`, `virtual` or `override` (except for next version of C#/.NET, only in interface)
- You have to call the static member with the `Base` class : `Console.Out.WriteLine(Base.X);`

**Comment by HaloFour** [comment]
See: https://github.com/dotnet/csharplang/issues/4436

This will enable you to declare `static abstract` members in an interface which can be used as a generic type constraint at which point you can call those static members from the generic type argument.  My understanding is that this will release in preview in C# 10.0 to get more feedback on the design as there are some known "gotchas".
